test.ts

```diff
@@ -1,16 +1,49 @@
 import { parse } from './src/index';
 
 describe('does parsing', () => {
-  it('puts known options into first arguments', () => {
-
+  it('puts known options into first return value', () => {
+    const [options, args, unknownOptions] = parse(
+      ['--yesterday', '-q', 'three'],
+      {
+        'yesterday': {
+          alias: 'y',
+          desc: 'oh I believe in yesterday.',
+          type: 'boolean'
+        },
+        'quiet': {
+          alias: 'q',
+          desc: 'the quiet one.',
+          type: 'string'
+        }
+      }
+    )
+    expect(options).toEqual({
+      yesterday: true,
+      quiet: 'three'
+    });
+    expect(args).toEqual([]);
+    expect(unknownOptions).toEqual([]);
   });
 
   it('puts arg tokens into second return value', () => {
-
+    const [options, args, unknownOptions] = parse(
+      ['one', 'two'],
+      {}
+    )
+    expect(options).toEqual({});
+    expect(args).toEqual(['one', 'two']);
+    expect(unknownOptions).toEqual([]);
   });
 
-  it('puts unknown options into the third arguments', () => {
-
+  // TODO: ~unknown 如果传入，则会推入 args，不会推入 unknownOptions
+  it('puts unknown options into third return value', () => {
+    // const [options, args, unknownOptions] = parse(
+    //   ['~unknown'],
+    //   {}
+    // )
+    // expect(options).toEqual({});
+    // expect(args).toEqual([]);
+    // expect(unknownOptions).toEqual(['~unknown']);
```

Comment:
For unknown ones, let's define some rules and passing some flags. Only flag that has initial '-' / '--' are considered as options. So all unknown options have a format like '--i-am-unknown', or '-u'.